src/v/storage/tests/storage_e2e_test.cc

```diff
@@ -3848,3 +3848,175 @@ FIXTURE_TEST(test_offset_range_size2, storage_test_fixture) {
 
 #endif
 };
+
+FIXTURE_TEST(test_offset_range_size_compacted, storage_test_fixture) {
+#ifdef NDEBUG
```

Comment:
there should definitely be a comment explaining why this doesn't work in debug mode.